test/controllers/api/v1/users_controller_test.rb

```diff
@@ -21,6 +21,46 @@ class Api::V1::UsersControllerTest < ActionDispatch::IntegrationTest
     assert_response :success
   end
 
+  test "exist returns /profile if the user is already registered" do
+    bob = create(:user, first_name: 'Bob', last_name: 'Belcher', email: 'bobs@burgers.com', zip: '62149', password: 'Archer' )
+    params = { user: { email: 'bobs@burgers.com'} }
+    post api_v1_users_exist_url, params: params, as: :json
```

Comment:
With the tests, a testing pattern we try to follow is to separate and group the _setup_, _actions_, and _expectations_.  

The way that would look for this test would be:

```ruby
test "exist returns /profile if the user is already registered" do
  bob = create(:user, first_name: 'Bob', last_name: 'Belcher', email: 'bobs@burgers.com', zip: '62149', password: 'Archer' )
  params = { user: { email: 'bobs@burgers.com'} }

  post api_v1_users_exist_url, params: params, as: :json
  json = response.parsed_body
  
  assert_equal '/profile', json['redirect_to']
end
```

Can you apply that testing pattern to all your specs.

Reply:
done.